react/src/components/atoms/buttons/ButtonWithIcon/ButtonWithIcon.stories.js

```diff
@@ -1,71 +1,134 @@
-        classes: array('classes', []),
-        iconSize: select('iconSize', buttonWithIconOptions.iconSize, ''),
-        size: select('size', buttonWithIconOptions.size, ''),
-        expanded: boolean('expanded', true),
-        capitalized: boolean('capitalized', true),
-        'arial-label': text('arial-label', ''),
-        'aria-haspopup': boolean('aria-haspopup', false),
-        theme: select('theme', buttonWithIconOptions.theme),
-        usage: select('usage', buttonWithIconOptions.usage)
-      };
-      // Capitalizes the name of each SVG icon to match
-      // what SVGR names components.
-      const component = select('name',
-        Object.fromEntries(
-          Object.entries(svgOptions).map(([key, value]) => [`Icon${key[0].toUpperCase() + key.slice(1)}`, value ? `Icon${value[0].toUpperCase() + value.slice(1)}` : value])
-          ),
-        'IconChevron'
-      );
-      const SelectedComponent = Icon[component];
-      const iconProps = {
-        width: 20,
-        height: 20
-      };
-      return(
-        <ButtonWithIcon {...props} icon={<SelectedComponent {...iconProps} />} />
-      );
-    }),
-    { info: ButtonWithIconDocs }
-  )
-  .add(
-    'ButtonSearch', (() => {
-      const icons = {
-        chevron: <Icon.IconChevron height={20} width={20} />,
-        search: <Icon.IconSearch height={20} width={20} />
-      };
-      const props = {
-        onClick: action('ButtonWithIcon clicked'),
-        usage: select('usage', buttonWithIconOptions.usage, '')
-      };
+export const ButtonWithIconExample = (args) => {
+  const {
+    name, width, height, ...rest
+  } = args;
+  const IconComponent = Icon[name];
+  return(<ButtonWithIcon {...rest} icon={<IconComponent width={width} height={height} />} />);
+};
+ButtonWithIconExample.parameters = {
+  docs: {
+    page: () => <StoryPage StoryComponent={ButtonWithIconExample} Description={ButtonWithIconDocs} />
+  }
+};
+ButtonWithIconExample.storyName = 'Default';
+ButtonWithIconExample.args = {
+  onClick: action('ButtonWithIcon clicked'),
+  text: 'Button With Icon',
+  type: 'submit',
+  classes: [],
+  iconSize: '',
+  size: '',
+  expanded: true,
+  capitalized: true,
+  'arial-label': '',
+  'aria-haspopup': false,
+  name: 'IconChevron',
+  width: 20,
+  height: 20
+};
+ButtonWithIconExample.argTypes = {
+  icon: {
+    control: {
+      disable: true
+    }
+  },
+  type: {
+    control: {
+      type: 'select',
+      options: buttonWithIconOptions.type
+    }
+  },
+  iconSize: {
+    control: {
+      type: 'select',
+      options: buttonWithIconOptions.iconSize
+    }
+  },
+  size: {
+    control: {
+      type: 'select',
+      options: buttonWithIconOptions.size
+    }
+  },
+  theme: {
+    control: {
+      type: 'select',
+      options: buttonWithIconOptions.theme
+    }
+  },
+  usage: {
+    control: {
+      type: 'select',
+      options: buttonWithIconOptions.usage
+    }
+  },
+  name: {
```

Comment:
this is a little confusing, as name is not a prop to the component but I get this is the only way will update the component in the story. 

Also this doesn't override the default prop Icon
![Screen Shot 2020-09-10 at 2 25 42 PM](https://user-images.githubusercontent.com/5789411/92781648-4fde2000-f372-11ea-8eb9-2045045021e0.png)

Reply:
maybe put iconName next to icon and just clear this is not a prop for the component?
Same comment for other components that uses Icon, e.g. CalloutAlert

Reply:
I made it so that `icon` is used instead of `name`.